### PR TITLE
ser: null-terminated strings

### PR DESCRIPTION
**Pull request by Namek**
#23

### CODE REVIEW

File: src/ser/impl/formatter/details/escape.zig [hunk 1]
```diff
@@ -93,7 +93,8 @@ pub fn escapeChar(codepoint: u21, writer: anytype) !void {
 }
 
 /// Escapes characters of a UTF-8 encoded string using JSON escape sequences.
-pub fn escape(bytes: []const u8, writer: anytype, formatter: anytype) !void {
+pub fn escape(input: []const u8, writer: anytype, formatter: anytype) !void {
+    var bytes = std.mem.sliceTo(input, 0);
```
Comment (ibokuri) [inline src/ser/impl/formatter/details/escape.zig:97]: Is it possible to make `bytes` `const` here? I don't believe it's modified anywhere in this function.

File: src/ser/serializer.zig [hunk 2]
```diff
@@ -119,12 +119,13 @@ pub fn Serializer(comptime Writer: type, comptime Formatter: type, comptime user
         }
 
         fn serializeString(self: *Self, value: anytype) Error!Ok {
-            if (!std.unicode.utf8ValidateSlice(value)) {
+            var slice = std.mem.sliceTo(value, 0);
```
Comment (ibokuri) [inline src/ser/serializer.zig:122]: Same here, can be this be made into a `const`?

**Review by ibokuri — CHANGES_REQUESTED** [review]

**Comment by phan-do** [comment]
Looks like this change is failing the test case for `"\u{0}"`, which should be serialized as `"\"\\u0000\""`, but is being serialized as `""` now.

**Comment by Namek** [comment]
closing, as discussed in #23

**Pull request closed without merging**